modules/browser_object_tabbar.py

```diff
@@ -0,0 +1,75 @@
+import logging
+from typing import Union
+
+from selenium.webdriver.remote.webelement import WebElement
+from selenium.webdriver.support import expected_conditions as EC
+
+from modules.page_base import BasePage
+
+
+class TabBar(BasePage):
+    """Page Object Model for tab navigation"""
+
+    URL_TEMPLATE = "about:blank"
+
+    class MediaStatus:
+        """Fake enum: just return a string based on a constant name"""
+
+        def __init__(self):
+            self.PLAYING = "playing"
+            self.MUTED = "muted"
+            self.AUTOPLAY_BLOCKED = "blocked"
+            self.PIP = "pip"
+
+    MEDIA_STATUS = MediaStatus()
+
+    def new_tab_by_button(self) -> BasePage:
+        """Use the New Tab button (+) to open a new tab"""
+        with self.driver.context(self.driver.CONTEXT_CHROME):
+            self.get_element("newtab-button").click()
+        return self
+
+    def click_tab_by_title(self, title: str) -> BasePage:
+        """Given a full page title, click the corresponding tab"""
+        with self.driver.context(self.driver.CONTEXT_CHROME):
+            self.get_element("tab-by-title", title).click()
+        return self
+
+    def click_tab_by_index(self, index: int) -> BasePage:
+        """Given a tab index (int), click the corresponding tab"""
+        with self.driver.context(self.driver.CONTEXT_CHROME):
+            self.get_element("tab-by-index", str(index)).click()
+        return self
+
+    def get_tab(self, identifier: Union[str, int]) -> Union[WebElement, None]:
+        """Return a tab root based on either a title or an index"""
+        with self.driver.context(self.driver.CONTEXT_CHROME):
+            if isinstance(identifier, int):
+                tab = self.get_element("tab-by-index", str(identifier))
+            elif isinstance(identifier, str):
+                tab = self.get_element("tab-by-title", identifier)
+            else:
+                # if we get an unexpected type, we shouldn't assume that the user wants sys exit
+                # but we have to cause problems for them nonetheless
+                assert False
```

Comment:
Q: What will this do; fail the test? Does it give some useful error message, or just a failed assert and sad trombone midi?

Reply:
It should fail the test but I see what you're getting at. These methods almost need their own tests, maybe even their own library... (foxpuppet)

Reply:
I would absolutely love to have a convo in future about how to implement foxpuppet here. I looked into it earlier in the repo's lifespan, but it did look kind of outdated. Maybe I just didn't understand it though!